ourjapanlife/findadoc-web
# refactor: add check for unsaved changes to modEditSubmissionForm

Related issue: #771

## 🔧 What changed
This PR refactors the `modEditSubmissionForm` component so that the modal only shows if a user hits the back button and has unsaved changes.

## 🧪 Testing instructions
In your local environment: 
1. Make a change to a submission and hit the back button. You should see the modal.
2. Make a change to a submission and hit the "Save and exit" button. You shouldn't see the modal.
3. Click a submission and hit the back button without making a change. You shouldn't see the modal.

See the steps in the README for how to log into the moderation panel.

## 📸 Screenshots
-   ### Before
![Screenshot 2024-10-10 at 1 30 53 AM](https://github.com/user-attachments/assets/44db6b71-366e-4385-a4ab-21b412b6a352)

-   ### After
![Screenshot 2024-10-10 at 1 29 16 AM](https://github.com/user-attachments/assets/0e1c3379-f9ac-484e-b978-e5cdb111d52a)
